app/Community/Controllers/ForumTopicController.php

```diff
@@ -59,7 +60,7 @@ public function create(ForumCategory $category, Forum $forum, Request $request):
     public function show(ForumTopic $topic, ShowForumTopicRequest $request): InertiaResponse|RedirectResponse
     {
         $this->authorize('view', $topic);
-        abort_if($topic->comments->isEmpty(), 404);
+        abort_if(!ForumTopicComment::where('forum_topic_id', $topic->id)->exists(), 404);
```

Comment:
This is indirectly related. The "comments" helper function queries all comments with their associated user data just to determine if records exist.

<img width="1485" height="387" alt="Image" src="https://github.com/user-attachments/assets/e61cbd4e-5824-490e-8969-af1914fd4179" />